Documentation/Books/Cookbook/AQL/Joins.md

```diff
@@ -359,194 +359,209 @@ arangosh> db.written.save("authors/2938210813",
 }
 ```
 
-In order to get all books with their authors you can use NEIGHBORS.
+In order to get all books with their authors you can use a [graph
+traversal](../../AQL/Graphs/Traversals.html#working-with-collection-sets)
 
 ```json
 arangosh> db._query(
-........>"FOR b IN books RETURN " + 
-........>"  { " +
-........>"    book: b, " +
-........>"    authors: NEIGHBORS(books," + 
-........>"                       written," + 
-........>"                       b._id," + 
-........>"                      'inbound'" +
-........>" ) }"
-........>).toArray();
-[ 
-  { 
-    "book" : { 
-      "_id" : "books/2980088317", 
-      "_rev" : "2980088317", 
-      "_key" : "2980088317", 
-      "title" : "The beauty of JOINS" 
-    }, 
-    "authors" : [ 
-      { 
-        "edge" : { 
-          "_id" : "written/3006237181", 
-          "_from" : "authors/2935261693", 
-          "_to" : "books/2980088317", 
-          "_rev" : "3006237181", 
-          "_key" : "3006237181", 
-          "pages" : "1-10" 
-        }, 
-        "vertex" : { 
-          "_id" : "authors/2935261693", 
-          "_rev" : "2935261693", 
-          "_key" : "2935261693", 
-          "name" : { 
-            "first" : "John", 
-            "last" : "Doe" 
-          } 
-        } 
-      }, 
-      { 
-        "edge" : { 
-          "_id" : "written/3012856317", 
-          "_from" : "authors/2938210813", 
-          "_to" : "books/2980088317", 
-          "_rev" : "3012856317", 
-          "_key" : "3012856317", 
-          "pages" : "11-20" 
-        }, 
-        "vertex" : { 
-          "_id" : "authors/2938210813", 
-          "_rev" : "2938210813", 
-          "_key" : "2938210813", 
-          "name" : { 
-            "first" : "Maxima", 
-            "last" : "Musterfrau" 
-          } 
-        } 
-      } 
-    ] 
-  } 
+...> "FOR b IN books " +
+...> "LET authorsByBook = ( " +
+...> "    FOR author IN INBOUND b._id written" +
```

Comment:
Please use:
`FOR author IN INBOUND b written OPTIONS {bfs: true, uniqueVertices: "global"} RETURN author`

As a query here. That is much more performant than the distinct variant. as the later cannot be optimized (yet)